test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -1576,7 +1576,9 @@ TEST_P(DnsImplTest, FallbackToNodataWithErrorOnAAAA) {
   EXPECT_NE(nullptr,
             resolveWithNoRecordsExpectation("some.good.domain", DnsLookupFamily::V4Preferred));
   dispatcher_->run(Event::Dispatcher::RunType::Block);
-  EXPECT_NE(nullptr, resolveWithNoRecordsExpectation("some.good.domain", DnsLookupFamily::All));
+  EXPECT_NE(nullptr,
+            resolveWithExpectations("some.good.domain", DnsLookupFamily::All,
+                                    DnsResolver::ResolutionStatus::Failure, {}, {}, absl::nullopt));
```

Comment:
Why this test change?

Reply:
OK I understand now. I think technically this is a behavior change then? I think it's probably fine, especially for ALL, but can you add a comment or maybe move this case to a different test since the title is FallbackToNodataWithErrorOnAAAA? Thank you.

/wait